neo/Ledger/Blockchain.cs

```diff
@@ -120,8 +121,8 @@ public class FillCompleted { }
         private readonly NeoSystem system;
         private readonly List<UInt256> header_index = new List<UInt256>();
         private uint stored_header_count = 0;
-        private readonly Dictionary<UInt256, Block> block_cache = new Dictionary<UInt256, Block>();
-        private readonly Dictionary<uint, LinkedList<Block>> block_cache_unverified = new Dictionary<uint, LinkedList<Block>>();
+        private readonly ConcurrentDictionary<UInt256, Block> block_cache = new ConcurrentDictionary<UInt256, Block>();
+        private readonly ConcurrentDictionary<uint, LinkedList<Block>> block_cache_unverified = new ConcurrentDictionary<uint, LinkedList<Block>>();
```

Comment:
If `block_cache_unverified` it's not used in different threads, we can revert this change and merge it